src/object/obj_ec.h

```diff
@@ -741,8 +772,16 @@ obj_ec_parity_lists_match(struct daos_recx_ep_list *lists_1,
 	for (i = 0; i < nr; i++) {
 		list_1 = &lists_1[i];
 		list_2 = &lists_2[i];
-		if (list_1->re_nr != list_2->re_nr ||
-		    list_1->re_ep_valid != list_2->re_ep_valid) {
+		if (list_1->re_ep_valid != list_2->re_ep_valid) {
+			D_ERROR("got different ep_valid in EC data recovery\n");
+			return -DER_IO;
+		}
+		if (list_1->re_nr != list_2->re_nr) {
+			if (obj_ec_parity_list_covered(list_1, list_2) ||
+			    obj_ec_parity_list_covered(list_2, list_1)) {
+				D_DEBUG(DB_IO, "parity list mismatch but fully covered\n");
```

Comment:
Hmm, server will set the epoch with vos_ec_agg, so the epoch should be same here?  I mean we should compare the epoch, if they are same, it can just use these extents, instead of keep FETCH_AGAIN to wait for the aggregation? no?

Reply:
if the recxs with same epoch, they should be already merged at server side before reply to client, see "daos_recx_ep_list_merge(parity_list, iods_nr)", so here at client-side, the different recxs should with different epoch already, right?

Reply:
ah, right.